packages/components/src/color-palette/test/index.tsx

```diff
@@ -236,24 +239,25 @@ describe( 'ColorPalette', () => {
 		);
 
 		// Confirm the correct color name, color value, and button label are used
-		expect( screen.getByText( EXAMPLE_COLORS[ 0 ].name ) ).toBeVisible();
-		expect( screen.getByText( EXAMPLE_COLORS[ 0 ].color ) ).toBeVisible();
+		expect(
+			screen.getByText( colorName, { selector: 'span' } )
```

Comment:
If you want to get a bit more specific, you can customize the `selector` by attribute:

```suggestion
			screen.getByText( colorName, { selector: 'span[data-wp-component="Truncate"]' } )
```

or by classname:

```suggestion
			screen.getByText( colorName, {
				selector: '.components-color-palette__custom-color-name',
			} )
```

or you can combine classname with tagname:

```suggestion
			screen.getByText( colorName, {
				selector: 'span.components-color-palette__custom-color-name',
			} )
```

Reply:
If we use your first suggestion, then we could put it into a variable as well: 

```
const truncateComponent = {
     selector: 'span[data-wp-component="Truncate"]',
};
```

But this made me wonder...

I am not 100% sure if the `Truncate` component will be kept in the upcoming `ColorPalette` changes. But if the parent is updated from a button to options, we can use a `getByRole` query, as I mentioned earlier. 

This made me think about tests in general: Is it best to specify the component so it would error if it is changed (so we remember to update to a better query), or is it better to use the class since it's more likely to stay the same and therefore, pass? 

I realize this is an implementation detail since it doesn't matter what we use as long as it ensures the behavior works for the user. I was just curious if you had an opinion on this. 🙂

Reply:
> If we use your first suggestion, then we could put it into a variable as well

We could if we need it multiple times, yes 👍 

> I am not 100% sure if the Truncate component will be kept in the upcoming ColorPalette changes. But if the parent is updated from a button to options, we can use a getByRole query, as I mentioned earlier.

`getByRole` is always preferred 👍 

> This made me think about tests in general: Is it best to specify the component so it would error if it is changed (so we remember to update to a better query), or is it better to use the class since it's more likely to stay the same and therefore, pass? I realize this is an implementation detail since it doesn't matter what we use as long as it ensures the behavior works for the user. I was just curious if you had an opinion on this. 

It really depends. In this case, we could argue that both the classname that I'm suggesting and the `span` that you are suggesting are implementation details. I guess in this case it's about what's the ideal implementation detail to include in the test. I believe the classname one is better, because there might be multiple span with the queried text and it would break the test; while it's a smaller chance to have multiple elements with the same text and the same classname. 

Anyway, I wanted to offer a few alternatives that seemed a bit better than just specifying a tagname, but feel free to ignore them, especially if we can change the query to be by role.

Reply:
Your suggestions are definitely better than just specifying a tag name, so thank you. And this answered my question! 

>I believe the classname one is better, because there might be multiple span with the queried text and it would break the test; while it's a smaller chance to have multiple elements with the same text and the same classname.